### PR TITLE
Receive addo referrals

### PR DESCRIPTION
**Pull request by maulidkategile**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -226,7 +226,7 @@ dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     testImplementation 'junit:junit:4.12'
 
-    implementation('org.smartregister:opensrp-client-core:1.8.31-DTREE-2-SNAPSHOT') {
+    implementation('org.smartregister:opensrp-client-core:1.8.31-DTREE-2-SNAPSHOT-LOCAL-01') {
```
Comment (gosso22) [inline app/build.gradle]: We need to publish this library and update here.

File: app/src/main/java/org/smartregister/addo/fragment/AddoVillageClientsFragment.java [hunk 2]
```diff
@@ -114,7 +115,7 @@ public void onChanged(String s) {
 
     @Override
     public void initializeAdapter(Set<org.smartregister.configurableviews.model.View> visibleColumns) {
-        AddoVillageClientsProvider addoVillageClientsProvider = new AddoVillageClientsProvider(getActivity(), paginationViewHandler, registerActionHandler, visibleColumns);
+        AddoVillageClientsProvider addoVillageClientsProvider = new AddoVillageClientsProvider(getActivity(), paginationViewHandler, registerActionHandler, visibleColumns, villageSelected);
```
Comment (gosso22) [inline app/src/main/java/org/smartregister/addo/fragment/AddoVillageClientsFragment.java:118]: What is the purpose of including villageSelected here, I am not sure I see where it is being used

File: app/src/main/java/org/smartregister/addo/provider/AddoVillageClientsProvider.java [hunk 3]
```diff
@@ -95,6 +101,15 @@ public void setAddressAndGender(CommonPersonObjectClient pc, RegisterViewHolder
         fillValue(viewHolder.textViewAddressAndGender, gender);
     }
 
+    public void setHasRefarral(CommonPersonObjectClient pc, RegisterViewHolder viewHolder){
+        if (ReferralUtils.hasReferralTask(CoreConstants.REFERRAL_PLAN_ID, LocationHelper.getInstance().getOpenMrsLocationId(villageSelected), pc.entityId(), CoreConstants.JsonAssets.LINKAGE_CODE)){
```
Comment (maulidkategile) [inline app/src/main/java/org/smartregister/addo/provider/AddoVillageClientsProvider.java]: @gosso22 Based on your comment https://github.com/Digital-Square-Tanzania/opensrp-client-addo/pull/28#discussion_r1587737116, the variable villageSelected represent CHW locationId and is used to fetch client linkage

**Comment by gosso22** [comment]
@issyzac and @maulidkategile - I think we need to close this PR as most of the changes are now into afyatek-ucs-integration branch and that is the one we will create a big PR against development or master branch. I will be waiting for your response then close it.

**Pull request closed without merging**